### DISCUSSION TITLE
Very Basic Planet with Detailed Subregion

### DISCUSSION THREAD

**Discussion by laneparton** [systemed/tilemaker#644] [Q&A] [answered]
I'm looking to generate a .mbtiles of Britain and Ireland and would love to support very basic shapes/outlines of the rest of the world/ocean but I'm having a difficult time understanding how to achieve this.

I've started testing with Luxembourg because its much smaller. I'm using the example resources from this repo - and I'm building v3 based on all the latest work being done in that branch.

```
docker run --rm -it \
  -v $(pwd)/generator:/srv \
  tilemaker \
    --input=/srv/luxembourg-latest.osm.pbf \
    --output=/srv/luxembourg-latest.mbtiles \
    --store=/srv/temp \
    --config=/srv/config-openmaptiles.json \
    --process=/srv/process-openmaptiles.lua
```

I've considered the following - some of these I've tried but some of them I've killed because waiting hours for a minor change seemed like a wasted effort. So - its possible I've killed a process that is actually worthy of seeing all the way through.

1. Load .shp files for the World from: https://www.naturalearthdata.com/downloads/10m-physical-vectors/ in landcover/landuse/ocean. I followed the simplify-below in the example JSON but I have no idea what this actually does.
2. Find a very small/minimal .mbtiles for the planet and merge it with my data
3. It seems like I need `"bounding_box": [-180,-60,180,75],` - when loading .shp files?

Without going into too much detail - is there an approach you suggest I go research? I'm not expecting an in-depth explanation, just something for me to go look into further :)

**Comment by laneparton** [comment]
I might have figured it out, just didn't give it enough time :) but I'd still love any best practices or insights. This is my first experience building map tiles.

I simplified it to Europe and let it run for a bit - and it seems to be doing what I expected.

```
{
	"layers": {
		"place":            { "minzoom":  0, "maxzoom": 14 },
		"boundary":         { "minzoom":  0, "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2 },

		"poi":              { "minzoom": 12, "maxzoom": 14 },
		"poi_detail":       { "minzoom": 14, "maxzoom": 14, "write_to": "poi"},

		"housenumber":      { "minzoom": 14, "maxzoom": 14 },

		"waterway":         { "minzoom":  8,  "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2 },
		"waterway_detail":  { "minzoom": 12,  "maxzoom": 14, "write_to": "waterway" },

		"transportation":             { "minzoom": 4,  "maxzoom": 14, "simplify_below": 13, "simplify_level": 0.0003 },
		"transportation_name":        { "minzoom": 8,  "maxzoom": 14 },
		"transportation_name_mid":    { "minzoom": 12, "maxzoom": 14, "write_to": "transportation_name" },
		"transportation_name_detail": { "minzoom": 14, "maxzoom": 14, "write_to": "transportation_name" },

		"building":          { "minzoom": 13, "maxzoom": 14 },

		"water":             { "minzoom": 6,  "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2},
		"ocean":             { "minzoom": 0,  "maxzoom": 14, "source": "/srv/resources/coastline/water_polygons.shp", "filter_below": 12,
		"simplify_below": 13,
		"simplify_level": 0.0002,
		"simplify_ratio": 2,
		"write_to": "water",
		"filter_area": 0.8  },
		"water_name":        { "minzoom": 14, "maxzoom": 14 },
		"water_name_detail": { "minzoom": 14, "maxzoom": 14, "write_to": "water_name" },

		"aeroway":           { "minzoom": 11, "maxzoom": 14 },
		"aerodrome_label":   { "minzoom": 10, "maxzoom": 14 },
		"park":              { "minzoom": 11, "maxzoom": 14 },
		"landuse":           { "minzoom":  4, "maxzoom": 14, "filter_area": 0.5,  "simplify_below": 13, "simplify_level": 0.0003, "simplify_ratio": 2 },
		"landcover":         { "minzoom":  0, "maxzoom": 14, "source": "/srv/resources/ne_10m_land/ne_10m_land.shp",    "filter_below": 12,
		"simplify_below": 13,
		"simplify_level": 0.0005,
		"simplify_ratio": 2,
		"filter_area": 1.0 },
		"mountain_peak":     { "minzoom": 11, "maxzoom": 14 }
	},
	"settings": {
		"minzoom": 0,
		"maxzoom": 14,
		"basezoom": 14,
		"include_ids": false,
		"combine_below": 14,
		"bounding_box": [-25.0, 34.5, 60.0, 71.0],
		"name": "Tilemaker to OpenMapTiles schema",
		"version": "3.0",
		"description": "Tile config based on OpenMapTiles schema",
		"compress": "gzip",
		"filemetadata": {
			"tilejson": "2.0.0", 
			"scheme": "xyz", 
			"type": "baselayer", 
			"format": "pbf", 
            "tiles": ["https://example.com/liechtenstein/{z}/{x}/{y}.pbf"]
		}
	}
}
```

**Comment by systemed** [comment, marked as answer]
Looks good!

For the use-case of "world coastline, road data UK-only" I think I'd start by producing an .mbtiles with just the coastline/oceans; you could do this by editing everything else out of the JSON/Lua config.

Then to add the UK to this, I'd use tilemaker's `--merge` option, basically superimposing the UK roads/rivers/etc. onto the coastline .mbtiles you've already produced.

(`--merge` modifies the .mbtiles in place, so you probably want to keep an unaltered copy somewhere which you can reuse. The coastline doesn't change very often ;) )

**Reply by laneparton** [reply]
Ooo I like that approach. I successfully ran it with Luxembourg this morning :)

Thanks for the insight - and I appreciate your efforts on Tilemaker!

**Reply by laneparton** [reply]
@systemed Quick question... What if I never want to degrade the UK? OpenStreetMaps always shows countries like the UK on all zoom levels. Are there parameters I can explore to replicate that experience with the tiles generated from Tilemaker??

![Screenshot 2024-01-16 at 3 41 56 PM](https://github.com/systemed/tilemaker/assets/2521591/16b9e7af-461e-4924-a1eb-69a71dc71d8e)

**Reply by systemed** [reply]
Sorry - I don't understand what you mean by "degrade".

**Reply by laneparton** [reply]
@systemed My apologies! That's not entirely clear. I'm unfamiliar with maps and tiling so bear with me :)

Basically, in the screenshot I shared - why is the UK missing completely at this Zoom level? Why is the end of Europe missing? Can I control that?

ChatGPT said:
> Zoom level 0 typically simplifies polygons and displays the world at a very coarse level for performance and usability reasons. This simplification is done to reduce the amount of data required to render the map at this zoom level, making it suitable for displaying the entire globe in a single tile. However, this simplification can lead to the loss of fine details, especially for smaller geographic features like countries, islands, and coastlines.

But I haven't gotten it to give me much help in changing that behavior hah.

I should note - that I've bound this .mbtiles to Europe so - I'm specifically asking about the map data within those bounds!

**Reply by systemed** [reply]
It looks to me that the geometry simplification setting is _way_ too high on that screenshot. If you do that then the geometries will break. Is your config the same one posted above?

**Reply by laneparton** [reply]
```
{
	"layers": {
		"place":            { "minzoom":  0, "maxzoom": 14 },
		"boundary":         { "minzoom":  0, "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2 },

		"poi":              { "minzoom": 12, "maxzoom": 14 },
		"poi_detail":       { "minzoom": 14, "maxzoom": 14, "write_to": "poi"},

		"housenumber":      { "minzoom": 14, "maxzoom": 14 },

		"waterway":         { "minzoom":  8,  "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2 },
		"waterway_detail":  { "minzoom": 12,  "maxzoom": 14, "write_to": "waterway" },

		"transportation":             { "minzoom": 4,  "maxzoom": 14, "simplify_below": 13, "simplify_level": 0.0003 },
		"transportation_name":        { "minzoom": 8,  "maxzoom": 14 },
		"transportation_name_mid":    { "minzoom": 12, "maxzoom": 14, "write_to": "transportation_name" },
		"transportation_name_detail": { "minzoom": 14, "maxzoom": 14, "write_to": "transportation_name" },

		"building":          { "minzoom": 13, "maxzoom": 14 },

		"water":             { "minzoom": 6,  "maxzoom": 14, "simplify_below": 12, "simplify_level": 0.0003, "simplify_ratio": 2},
		"ocean":             { "minzoom": 0,  "maxzoom": 14, "source": "/srv/resources/ne_10m_ocean.shp", "simplify_below": 14, "simplify_level": 0.0002, "simplify_ratio": 2, "write_to": "water"  },
		"coastline":             { "minzoom": 0,  "maxzoom": 14, "source": "/srv/resources/ne_10m_coastline.shp", "simplify_below": 14, "simplify_level": 0.0002, "simplify_ratio": 2, "write_to": "water"  },
		"water_name":        { "minzoom": 14, "maxzoom": 14 },
		"water_name_detail": { "minzoom": 14, "maxzoom": 14, "write_to": "water_name" },

		"aeroway":           { "minzoom": 11, "maxzoom": 14 },
		"aerodrome_label":   { "minzoom": 10, "maxzoom": 14 },
		"park":              { "minzoom": 11, "maxzoom": 14 },
		"landuse":           { "minzoom":  4, "maxzoom": 14, "simplify_below": 13, "simplify_level": 0.0003, "simplify_ratio": 2 },
		"urban_areas":       { "minzoom":  4, "maxzoom":  8, "source": "landcover/ne_10m_urban_areas/ne_10m_urban_areas.shp", "source_columns": ["featurecla"], "simplify_below": 7, "simplify_level": 0.0003, "simplify_ratio": 2, "write_to": "landuse" },
		"landcover":         { "minzoom":  0, "maxzoom": 14, "simplify_below": 13, "simplify_level": 0.0003, "simplify_ratio": 2 },
		"ice_shelf":         { "minzoom":  0, "maxzoom":  9, "source": "landcover/ne_10m_antarctic_ice_shelves_polys/ne_10m_antarctic_ice_shelves_polys.shp", "source_columns": ["featurecla"], "simplify_below": 13, "simplify_level": 0.0005, "write_to": "landcover" },
		"glacier":           { "minzoom":  2, "maxzoom":  9, "source": "landcover/ne_10m_glaciated_areas/ne_10m_glaciated_areas.shp", "source_columns": ["featurecla"], "simplify_below": 13, "simplify_level": 0.0005, "write_to": "landcover" },
		"mountain_peak":     { "minzoom": 11, "maxzoom": 14 }
	},
	"settings": {
		"minzoom": 0,
		"maxzoom": 14,
		"basezoom": 14,
		"include_ids": false,
		"combine_below": 14,
		"bounding_box": [-25.0, 34.5, 60.0, 71.0],
		"name": "Tilemaker to OpenMapTiles schema",
		"version": "3.0",
		"description": "Tile config based on OpenMapTiles schema",
		"compress": "gzip",
		"filemetadata": {
			"tilejson": "2.0.0", 
			"scheme": "xyz", 
			"type": "baselayer", 
			"format": "pbf", 
            "tiles": ["https://example.com/liechtenstein/{z}/{x}/{y}.pbf"]
		}
	}
}
```

Basically I grabbed this: https://github.com/systemed/tilemaker/blob/v3/resources/config-openmaptiles.json

And tried to use the NaturalEarth data for Coastlines/Oceans but I've done so much trial and error that it probably differs slightly from the reference JSON :)

**Reply by systemed** [reply]
Ok. I'd suggest you use the coastline files from osmdata.openstreetmap.de as mentioned in the README, and use the ocean line verbatim from config-openmaptiles.json. You have

{ "minzoom": 0,  "maxzoom": 14, "source": "/srv/resources/ne_10m_ocean.shp", "simplify_below": 14, "simplify_level": 0.0002, "simplify_ratio": 2, "write_to": "water"  }

whereas config-openmaptiles.json has

{ "minzoom": 0,  "maxzoom": 14, "source": "coastline/water_polygons.shp", "filter_below": 12, "filter_area": 0.5, "simplify_below": 13, "simplify_level": 0.0001, "simplify_ratio": 2, "write_to": "water" }

**Reply by laneparton** [reply]
Much better - my apologies for straying off the README! Thanks for your help @systemed

**Reply by systemed** [reply]
No problem!

**Comment by systemed** [comment]
This is quite a common request so I've added documentation and sample config in #648.
